plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/partition/HudiPartitionInfoLoader.java

```diff
@@ -73,6 +87,7 @@ private void generateSplitsFromPartition(String partitionName)
                 List<HivePartitionKey> partitionKeys = hudiPartitionInfo.getHivePartitionKeys();
                 List<FileSlice> partitionFileSlices = hudiDirectoryLister.listStatus(hudiPartitionInfo, commitTime);
                 partitionFileSlices.stream()
+                        .filter(slice -> indexSupportOpt.map(hudiIndexSupport -> hudiIndexSupport.shouldKeepFileSlice(slice)).orElse(true))
```

Comment:
Could the `indexSupportOpt` be passed to the constructor of `HudiSnapshotDirectoryLister` so `hudiDirectoryLister` does data skipping with `#listStatus`?

Reply:
Data skipping is part of the file listing conceptually; the split and partition info loaders are only responsible for constructing and returning splits.